plugins/repository-hdfs/src/javaRestTest/java/org/elasticsearch/repositories/hdfs/HaHdfsFailoverTestSuiteIT.java

```diff
@@ -8,271 +8,46 @@
-            for (HAServiceProtocol protocol : protocolsToClose) {
-                if (protocol instanceof HAServiceProtocolClientSideTranslatorPB haServiceProtocolClientSideTranslatorPB) {
-                    haServiceProtocolClientSideTranslatorPB.close();
-                }
-            }
-        }
-    }
-
-    /**
-     * The default HAAdmin tool does not throw exceptions on failures, and does not close any client connection
-     * resources when it concludes. This subclass overrides the tool to allow for exception throwing, and to
-     * keep track of and clean up connection resources.
-     */
-    private static class CloseableHAAdmin extends DFSHAAdmin {
-        private final List<CloseableHAServiceTarget> serviceTargets = new ArrayList<>();
-
-        @Override
-        protected HAServiceTarget resolveTarget(String nnId) {
-            CloseableHAServiceTarget target = new CloseableHAServiceTarget(super.resolveTarget(nnId));
-            serviceTargets.add(target);
-            return target;
-        }
-
-        @Override
-        public int run(String[] argv) throws Exception {
-            return runCmd(argv);
-        }
-
-        public int transitionToStandby(String namenodeID) throws Exception {
-            return run(new String[] { "-transitionToStandby", namenodeID });
-        }
-
-        public int transitionToActive(String namenodeID) throws Exception {
-            return run(new String[] { "-transitionToActive", namenodeID });
-        }
-
-        public void close() {
-            for (CloseableHAServiceTarget serviceTarget : serviceTargets) {
-                serviceTarget.close();
-            }
-        }
+    @Override
+    HdfsFixture getHdfsFixture() {
+        return hdfsFixture;
     }
 
-    /**
-     * Performs a two-phase leading namenode transition.
-     * @param from Namenode ID to transition to standby
-     * @param to Namenode ID to transition to active
-     * @param configuration Client configuration for HAAdmin tool
-     * @throws IOException In the event of a raised exception during namenode failover.
-     */
-    private void failoverHDFS(String from, String to, Configuration configuration) throws IOException {
-        logger.info("Swapping active namenodes: [{}] to standby and [{}] to active", from, to);
-        try {
-            AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
-                CloseableHAAdmin haAdmin = new CloseableHAAdmin();
-                haAdmin.setConf(configuration);
-                try {
-                    haAdmin.transitionToStandby(from);
-                    haAdmin.transitionToActive(to);
-                } finally {
-                    haAdmin.close();
-                }
-                return null;
-            });
-        } catch (PrivilegedActionException pae) {
-            throw new IOException("Unable to perform namenode failover", pae);
-        }
+    @Override
+    String securityCredentials() {
+        return "";
```

Comment:
Maybe we can just make this the default in `AbstractHaHdfsfailoverTestSuiteIT` and only override in the secure test case?

Reply:
done